inst/shiny-server/server/ERTPlot.R

```diff
@@ -111,28 +187,38 @@ render_ERTPlot_aggr_plot <- reactive({
     shinyjs::alert("This plot is only available when the dataset contains multiple algorithms for the selected dimension.")
     return(NULL)
   }
-  erts <- MAX_ERTS_FUNC()
+  data
+}
+
+default_targets_table <- reactive({
+  data <- ERTPlot.Aggr.data()
+  if (is.null(data)) return(NULL)
   aggr_on <- 'funcId'
-  aggr_attr <- if (aggr_on == 'funcId') get_funcId(data) else get_DIM(data)
-  targets <- get_max_targets(data, aggr_on, maximize = !(format == COCO || format == BIBOJ_COCO))
-  erts <- max_ERTs(data, aggr_on, targets, maximize = !(format == COCO || format == BIBOJ_COCO))
-
-  Plot.RT.Aggregated(data, plot_mode = input$ERTPlot.Aggr.Mode, targets = targets,
-                scale.ylog = input$ERTPlot.Aggr.Logy,
-                maximize = attr(data[[1]],'maximization'),
-                use_rank = input$ERTPlot.Aggr.Ranking,
-                aggr_on = aggr_on, erts = erts)
-  },
-  message = "Creating plot")
+  get_max_targets(data, aggr_on, maximize = attr(data[[1]],'maximization')) %>% t %>% as.data.table(keep.rownames = F)
+})
+
+ERTPlot.Aggr.Targets_obj <- NULL
+
+proxy_ERTPlot.Aggr.Targets <- dataTableProxy('ERTPlot.Aggr.Targets')
+
+output$ERTPlot.Aggr.Targets <- DT::renderDataTable({
+  req(length(DATA_RAW()) > 0)
+  ERTPlot.Aggr.Targets_obj <<- default_targets_table()
+  ERTPlot.Aggr.Targets_obj
+}, editable = TRUE, rownames = FALSE,
+options = list(pageLength = 5, lengthMenu = c(5, 10, 25, -1), scrollX = T, server = T))
+
+
+observeEvent(input$ERTPlot.Aggr.Targets_cell_edit, {
```

Comment:
I don't recall this control widget id: ERTPlot.Aggr.Targets_cell_edit. We had some new functionalities here?

Reply:
These are automatically generated widget ids, created by the editable data table to be able to observe changes made by the user